pkg/asset/machines/baremetal/hosts.go

```diff
@@ -34,76 +93,37 @@ func Hosts(config *types.InstallConfig, machines []machineapi.Machine) (*HostSet
 		return nil, fmt.Errorf("no baremetal platform in configuration")
 	}
 
-	for i, host := range config.Platform.BareMetal.Hosts {
-		bmc := baremetalhost.BMCDetails{}
-		if host.BMC.Username != "" && host.BMC.Password != "" {
-			// Each host needs a secret to hold the credentials for
-			// communicating with the management controller that drives
-			// that host.
-			secret := corev1.Secret{
-				TypeMeta: metav1.TypeMeta{
-					APIVersion: "v1",
-					Kind:       "Secret",
-				},
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      fmt.Sprintf("%s-bmc-secret", host.Name),
-					Namespace: "openshift-machine-api",
-				},
-				Data: map[string][]byte{
-					"username": []byte(host.BMC.Username),
-					"password": []byte(host.BMC.Password),
-				},
-			}
-			bmc.Address = host.BMC.Address
-			bmc.CredentialsName = secret.Name
-			bmc.DisableCertificateVerification = host.BMC.DisableCertificateVerification
-			settings.Secrets = append(settings.Secrets, secret)
-		}
+	numRequiredMasters := len(machines)
+	numMasters := 0
+	for _, host := range config.Platform.BareMetal.Hosts {
 
-		// Map string 'default' to hardware.DefaultProfileName
-		if host.HardwareProfile == "default" {
-			host.HardwareProfile = hardware.DefaultProfileName
+		secret, bmc := createSecret(host)
+		if secret != nil {
+			settings.Secrets = append(settings.Secrets, *secret)
 		}
+		newHost := createBaremetalHost(host, bmc)
```

Comment:
Should newHost be created inside the check for if we have insufficient masters later in https://github.com/openshift/installer/pull/5205/files#diff-ae5c907c1b30aadff3caf9124cfc777ab5a253bf12b56b63e4b2f428132c9022R106?

Reply:
For that I kept the the same previous logic: here we're preparing the data for generating all the baremetal related manifests (secrets, masters and moreover workers). The assumption to have enough hosts is guaranteed by the previous validation rules applied by the `InstallConfig` asset